numpyro/distributions/continuous.py

```diff
@@ -1075,6 +1075,145 @@ def cdf(self, x):
         return self.base_dist.cdf(jnp.log(x))
 
 
+def _batch_solve_triangular(A, B, sample_shape, batch_shape, event_shape):
+    """Extende solve_triangular for the case that B.ndim > A.ndim.
+        This is achived by first flattening the leadind B.ndim - A.ndim axes of B and
+        then moving them to the end.
+
+    Args:
+        A (_type_): A triangular array.
+        B (_type_): _description_
+        sample_shape (_type_): _description_
+        batch_shape (_type_): _description_
+        event_shape (_type_): _description_
+
+    Returns:
+        _type_: _description_
+    """
+    n, p = event_shape
+    B_flat = jnp.moveaxis(B.reshape((-1,) + batch_shape + event_shape), 0, -2).reshape(
```

Comment:
Can we broadcast `B` to `sample_shape + batch_shape + event_shape` before performing `reshape`? Most of the time it will be a no-op but might be convenient if we have a batched distribution but only wants to evaluate log_prob over a non-batched sample.